src/Engine.php

```diff
@@ -61,10 +61,12 @@ public function update($models)
             return;
         }
 
-        $index = $models->first()->searchableAs();
+        $model = $models->first();
+        $index = $model->searchableAs();
+        $routingPath = in_array('ElasticScoutDriverPlus\ShardRouting', class_uses_recursive($model)) ? $model->getRoutingPath() : null;
```

Comment:
I don't like, that we couple two packages here. Ideally, Scout Driver shouldn't know about Plus package. Maybe `method_exists($model, 'getRoutingPath')` is more appropriate in this context?

Reply:
I agree but i couldn't find any other way :)
Your suggestion would work but imho is more error prone and dangerous.

I think that our only truly viable solution is to override engine in plus. Seems like code duplication now but i can imagine more non scout supported features will be needed and having an engine in plus could really simplify the development of the packages

Reply:
Another clean solution would be to return to the initial idea of having the documents hold the routing path.
Then we could maybe override DocumentFactory::makeFromModels in plus and let elastic-adapter handle the routing path.

Hmmm..
Another option could ( writing while thinking :P )

Make DocumentFactory do more since the DocumentFactory is the only one that should know about models.
So what if we had a DocumentFactory::getExtras($models) that would return an empty array on scout-driver and a 
```
[
 'routing_path' => $models->first()->getRoutingPath()
 ]
 ```
 on plus.
 
 We would then need to update index and delete to take a fourth array argument ( instead of routingPath) which on document manager level would do apply additional logic.

Reply:
I kinda like the DocumentFactory idea, would serve as backdoor for plus so that we keep DocumentManager unaware of models.

Reply:
I'd go for a more straightforward solution. We already have `EngineDecorator` in Plus package, so maybe we can just add two additional methods there (`update` and `delete`). It's not much of a duplication as the main functionality is extracted in `DocumentManager`.

Reply:
Or u have something else in your mind?

Reply:
Yes, I don't see a better option atm and you right, we might some extra functionality in Plus version in the future, so let's better not limit ourselves.